flaky test: TestCompact

```
=== RUN   TestCompact
2024/01/16 03:02:16.317865 juicefs[25718] <INFO>: Meta address: redis://127.0.0.1:6379/11 [interface.go:521]
2024/01/16 03:02:16.318472 juicefs[25718] <WARNING>: AOF is not enabled, you may lose data if Redis is not shutdown properly. [info.go:84]
2024/01/16 03:02:16.318584 juicefs[25718] <INFO>: Ping redis latency: 69.389µs [redis.go:3624]
2024/01/16 03:02:16.318879 juicefs[25718] <INFO>: Data use file:///tmp/TestCompact2411249756/001/test/ [format.go:471]
2024/01/16 03:02:16.319494 juicefs[25718] <INFO>: Volume is formatted as {
  "Name": "test",
  "UUID": "ac8a52de-a656-4b10-bfd4-ab6b35c1366c",
  "Storage": "file",
  "Bucket": "/tmp/TestCompact2411249756/001/",
  "BlockSize": 4096,
  "Compression": "none",
  "EncryptAlgo": "aes256gcm-rsa",
  "TrashDays": 0,
  "MetaVersion": 1,
  "MinClientVersion": "1.1.0-A",
  "DirStats": true
} [format.go:508]
2024/01/16 03:02:16.321487 juicefs[25718] <INFO>: Meta address: redis://127.0.0.1:6379/11 [interface.go:521]
2024/01/16 03:02:16.322056 juicefs[25718] <WARNING>: AOF is not enabled, you may lose data if Redis is not shutdown properly. [info.go:84]
2024/01/16 03:02:16.322150 juicefs[25718] <INFO>: Ping redis latency: 58.92µs [redis.go:[362](https://github.com/juicedata/juicefs/actions/runs/7536446110/job/20513880441?pr=4335#step:9:363)4]
2024/01/16 03:02:16.322654 juicefs[25718] <INFO>: Data use file:///tmp/TestCompact2411249756/001/test/ [mount.go:647]
2024/01/16 03:02:16.322807 juicefs[25718] <INFO>: Disk cache (/var/jfsCache/ac8a52de-a656-4b10-bfd4-ab6b35c1[366](https://github.com/juicedata/juicefs/actions/runs/7536446110/job/20513880441?pr=4335#step:9:367)c/): capacity (102400 MB), free ratio (10%), max pending pages (15) [disk_cache.go:116]
2024/01/16 03:02:16.323[370](https://github.com/juicedata/juicefs/actions/runs/7536446110/job/20513880441?pr=4335#step:9:371) juicefs[25718] <INFO>: Create session 1 OK with version: 1.2.0-dev+unknown [base.go:494]
2024/01/16 03:02:16.323802 juicefs[25718] <INFO>: Prometheus metrics listening on 127.0.0.1:38327 [mount.go:158]
2024/01/16 03:02:16.323931 juicefs[25718] <INFO>: Mounting volume test at /tmp/jfs-unit-test ... [mount_unix.go:290]
2024/01/16 03:02:16.823846 juicefs[25718] <INFO>: OK, test is ready at /tmp/jfs-unit-test [mount_unix.go:57]
    mount_test.go:142: mount /tmp/jfs-unit-test success
2024-01-16 03:02:19.323244 I | Unimplemented opcode POLL
2024/01/16 03:02:19.407609 juicefs[25718] <WARNING>: Transaction succeeded after 3 tries (5.013884ms), keys: [sliceRef], last error: redis: transaction failed [redis.go:917]
2024/01/16 03:02:19.408382 juicefs[25718] <WARNING>: Transaction succeeded after 3 tries (5.51324ms), keys: [sliceRef], last error: redis: transaction failed [redis.go:917]
2024/01/16 03:02:19.410731 juicefs[25718] <WARNING>: Transaction succeeded after 3 tries (7.210228ms), keys: [sliceRef], last error: redis: transaction failed [redis.go:917]
2024/01/16 03:02:19.417517 juicefs[25718] <WARNING>: Transaction succeeded after 3 tries (8.48611ms), keys: [sliceRef], last error: redis: transaction failed [redis.go:917]
2024/01/16 03:02:19.425521 juicefs[25718] <WARNING>: Transaction succeeded after 3 tries (5.503629ms), keys: [sliceRef], last error: redis: transaction failed [redis.go:917]
2024/01/16 03:02:19.427219 juicefs[25718] <WARNING>: Transaction succeeded after 3 tries (6.724072ms), keys: [sliceRef], last error: redis: transaction failed [redis.go:917]
2024/01/16 03:02:19.702277 juicefs[25718] <INFO>: compact [2:/tmp/jfs-unit-test/d1] success. [compact_unix.go:129]
    compact_test.go:116: 
        	Error Trace:	/home/runner/work/juicefs/juicefs/cmd/compact_test.go:116
        	Error:      	Not equal: 
        	            	expected: 60
        	            	actual  : 50
        	Test:       	TestCompact
2024/01/16 03:02:20.005655 juicefs[25718] <INFO>: compact [34:/tmp/jfs-unit-test/d2] success. [compact_unix.go:129]
    compact_test.go:116: 
        	Error Trace:	/home/runner/work/juicefs/juicefs/cmd/compact_test.go:116
        	Error:      	Not equal: 
        	            	expected: 45
        	            	actual  : 35
        	Test:       	TestCompact
2024/01/16 03:02:20.309327 juicefs[25718] <INFO>: compact [3:/tmp/jfs-unit-test/d1/d11] success. [compact_unix.go:129]
2024/01/16 03:02:20.311073 juicefs[25718] <INFO>: close session 1: %!s(<nil>) [base.go:621]
2024/01/16 03:02:20.311107 juicefs[25718] <INFO>: The juicefs mount process exit successfully, mountpoint: /tmp/jfs-unit-test [mount.go:701]
--- FAIL: TestCompact (4.00s)
FAIL
	github.com/juicedata/juicefs/cmd	coverage: 25.2% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/chunk	coverage: 48.7% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/compress	coverage: 23.7% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/fs	coverage: 0.0% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/fuse	coverage: 52.8% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/gateway	coverage: 0.0% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/meta	coverage: 12.9% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/metric	coverage: 18.0% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/object	coverage: 5.7% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/sync	coverage: 1.1% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/usage	coverage: 0.0% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/utils	coverage: 53.5% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/version	coverage: 65.1% of statements in ./pkg/..., ./cmd/...
	github.com/juicedata/juicefs/pkg/vfs	coverage: 44.2% of statements in ./pkg/..., ./cmd/...
FAIL	github.com/juicedata/juicefs/cmd	10.110s
FAIL
make: *** [Makefile:89: test.cmd] Error 1
Error: Process completed with exit code 2.
```

--- a/cmd/compact_test.go
+++ b/cmd/compact_test.go
@@ -23,7 +23,6 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 )
 
@@ -71,7 +70,6 @@ func initForCompactTest(mountDir string, dirs map[string]testDir) {
 }
 
 func TestCompact(t *testing.T) {
-	logger.Level = logrus.DebugLevel
 	var bucket string
 	mountTemp(t, &bucket, []string{"--trash-days=0"}, nil)
 	defer umountTemp(t)
@@ -107,7 +105,10 @@ func TestCompact(t *testing.T) {
 	chunkCnt := getFileCount(dataDir)
 	assert.Equal(t, sumChunks, chunkCnt)
 
-	for _, d := range dirs {
+	orderedDirs := []string{"d1/d11", "d1", "d2"}
+	for _, path := range orderedDirs {
+		d := dirs[path]
+
 		err := Main([]string{"", "compact", filepath.Join(testMountPoint, d.path)})
 		assert.Nil(t, err)
 

